app/scripts/lib/validate.js

```diff
@@ -28,28 +44,27 @@ define(function (require, exports, module) {
      * @return true if email is valid, false otw.
      */
     isEmailValid: function (email) {
-      if (typeof email !== 'string') {
+      if (typeof email !== 'string' || email.length > 256) {
         return false;
       }
 
       var parts = email.split('@');
+      var localPart = parts[0];
+      var domainPart = parts[1];
+
+      if (parts.length !== 2 || localPart.length > 64) {
+        return false;
+      }
+
+      if (unicodeRegex.test(localPart)) {
+        localPart = punycode.toASCII(localPart);
```

Comment:
> ...they fail in 4.3. I'm still looking into why...

Fwiw, this failure was just because the mailbox test helper wasn't URI-encoding the email address, as per https://github.com/mozilla/fxa-auth-server/pull/1207.